app/livechat/client/views/app/tabbar/visitorInfo.js

```diff
@@ -84,7 +79,31 @@ Template.visitorInfo.helpers({
 					const customFields = Template.instance().customFields.get();
 					if (customFields) {
 						const field = _.findWhere(customFields, { _id });
-						if (field && field.visibility !== 'hidden') {
+						if (field && field.visibility !== 'hidden' && field.scope === 'room') {
+							fields.push({ label: field.label, value: livechatData[_id] });
+						}
+					}
+				}
+			}
+			return fields;
+		}
+	},
+
+	customVisitorFields() {
+		const fields = [];
+		let livechatData = {};
+		const user = Template.instance().user.get();
+		if (user) {
+			livechatData = _.extend(livechatData, user.livechatData);
+		}
+
+		if (!_.isEmpty(livechatData)) {
+			for (const _id in livechatData) {
```

Comment:
And Here, why not use `Object.keys` instead of these `if's` ?
You can do something like this:

```
		Object.keys(livechatData).forEach((key) => {
			const field = _.findWhere(customFields, { key });
			...
		});
```